core/engine_test.go

```diff
@@ -1127,7 +1159,9 @@ func TestEngineRunsTeardownEvenAfterTestRunIsAborted(t *testing.T) {
 		VUs: null.IntFrom(1), Iterations: null.IntFrom(1),
 	})
 
-	assert.NoError(t, run())
+	// Act
+	err := run()
+	require.NoError(t, err)
```

Comment:
```
assert.NoError(t, run())
```
we often use this form in the codebase, you think that is it not readable?

Reply:
For me, the single line variant is much more readable... It's the test equivalent of `if err := something; err != nil {}`. Why would we have `err` in a global variable if we're not going to use it down the line?

Reply:
As in my previous comment: I wholeheartedly disagree with you folks, but I don't have a strong enough opinion about it to argue 😄

This change was inline with `AAA` testing, and I should probably have renamed `err` to `gotErr` to be more explicit. Personally I find it a better practice, because it's more explicit, to be as explicit and stateful as possible in the context of a test.

Regarding the `if err := something; err != nil {}` format, I think it's a matter of personal preference, and I don't care enough to argue about it, but personally I find this syntax not so readable and always avoid it in my code. Once again it's personal and likely linked to our own experiences, but I do find:

```go
err := something()
if err != nil {
}
```

Reply:
Explicit > shorter is fine, but I _personally_ prefer to have each variable live in the smallest possible scope, whenever possible, to avoid "pollution". It is also easier to reason about, at least for me, because with `require.NoError(t, someFunc())` and `if err := something; err != nil {}` I don't need to wonder if `err` will be used below.